### ISSUE TITLE
NZBHydra fails for java version detection (OpenJDK 16)

### ISSUE THREAD

**Issue by SchlauFuchs** [theotherp/nzbhydra2#751]
Hi, This is the log when starting NZBHydra2 with the console executable on windows:

```C:\Program Files\nzbhydra>"NZBHydra2 Console.exe"
Logging wrapper output to C:\Program Files\nzbhydra\data\logs\wrapper.log
2022-01-07 22:24:52,240  INFO - Determined java version as '16' from version string 'openjdk version "16" 2021-03-16'
2022-01-07 22:24:52,243  INFO - Starting NZBHydra main process with command line: java -Xmx256M -DfromWrapper -XX:TieredStopAtLevel=1 -noverify -XX:+HeapDumpOnO
utOfMemoryError -XX:HeapDumpPath=C:\Program Files\nzbhydra\data\logs -jar C:\Program Files\nzbhydra\lib\core-4.0.0-exec.jar --datafolder C:\Program Files\nzbhyd
ra\data in folder C:\Program Files\nzbhydra
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
22:24:52.835 [main] ERROR org.nzbhydra.NzbHydra - Unable to determine java version
java.lang.NumberFormatException: null
        at java.base/java.lang.Integer.parseInt(Integer.java:622)
        at java.base/java.lang.Integer.parseInt(Integer.java:778)
        at org.nzbhydra.NzbHydra.checkJavaVersion(NzbHydra.java:136)
        at org.nzbhydra.NzbHydra.main(NzbHydra.java:93)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
22:24:52.863 [main] ERROR org.nzbhydra.NzbHydra - NZBHydra 2 may not work properly when run your windows program files folder. Please put it somewhere else
2022-01-07 22:24:52,900  ERROR - Main process shut down unexpectedly. If the wrapper was started in daemon mode you might not see the error output. Start Hydra manually with the same parameters in the same environment to see it```

```C:\Program Files\nzbhydra>java --version
openjdk 16 2021-03-16
OpenJDK Runtime Environment (build 16+36-2231)
OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
```

**Comment by theotherp** [comment]
Argh. Why is it so fucking hard to parse that.
I'm unable to reproduce that using v16 from https://developer.ibm.com/languages/java/semeru-runtimes/downloads or http://jdk.java.net/java-se-ri/16
Where did you get that from?

I recommend you install Java 17 from https://adoptium.net/.

**Comment by pb1051** [comment]
I have a similar issue, but running in a docker container.

**Comment by theotherp** [comment]
By which maintainer?

**Comment by edzeg** [comment]
> I have a similar issue, but running in a docker container.

*tips hat at fellow CB user*

> By which maintainer?

It's Hotio's container:
https://hotio.dev/containers/nzbhydra2/

**Comment by pb1051** [comment]
I deleted some files taking up disk space and then restored from a backup.
Seems to be working fine again.

**Comment by theotherp** [comment]
Then it can't have been the Java executable.

**Comment by pb1051** [comment]
Yah, that's part of the error message, but the solution dealt with memory.
